api/v1beta2/awscluster_webhook.go

```diff
@@ -42,21 +43,31 @@ const (
 var _ = ctrl.Log.WithName("awscluster-resource")
 
 func (r *AWSCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	w := new(awsClusterWebhook)
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
+		WithValidator(w).
```

Comment:
We could take this opportunity to move the webhooks to `internal/webhooks` to be more in line with upstream cluster API and also kubebuilder.

Reply:
As was mentioned in the CAPG PR, let's defer the moving of the code.